qa/sc_evm_node_info.py

```diff
@@ -0,0 +1,201 @@
+            sc_node_1_configuration,
+            sc_node_2_configuration
+        )
+
+        # rewind sc genesis block timestamp for 5 consensus epochs
+        self.sc_nodes_bootstrap_info = bootstrap_sidechain_nodes(self.options, network,
+                                                                 block_timestamp_rewind=self.block_timestamp_rewind,
+                                                                 model=AccountModel)
+
+    def run_test(self):
+        time.sleep(0.1)
+
+        ft_amount_in_zen = 10
+        ft_amount_in_zennies = convertZenToZennies(ft_amount_in_zen)
+        ft_amount_in_wei = convertZenniesToWei(ft_amount_in_zennies)
+
+        self.sc_ac_setup(ft_amount_in_zen=ft_amount_in_zen)
+        mc_node = self.nodes[0]
+        sc_node1 = self.sc_nodes[0]
+        sc_node2 = self.sc_nodes[1]
+
+        connect_sc_nodes(sc_node1, 1)  # Connect SC nodes
+
+        assert_true(sc_node1.submitter_isCertificateSubmitterEnabled()["result"]["enabled"],
+                    "sc node1 should be submitter")
+        assert_false(sc_node2.submitter_isCertificateSubmitterEnabled()["result"]["enabled"],
+                     "sc node2 should not be submitter")
+
+        hex_evm_addr = remove_0x_prefix(self.evm_address)
+
+        # Checks that FT appears in SC account balance
+        new_balance = http_wallet_balance(sc_node1, hex_evm_addr)
+        assert_equal(new_balance, ft_amount_in_wei, "wrong balance")
+
+        # Generate 8 more MC block to finish the first withdrawal epoch, then generate 1 more SC block to sync with MC.
+        we0_end_mcblock_hash = mc_node.generate(8)[7]
+
+        scblock_id2 = generate_next_block(sc_node1, "first node")
+        check_mcreferencedata_presence(we0_end_mcblock_hash, scblock_id2, sc_node1)
+
+        # Generate first mc block of the next epoch
+        we1_1_mcblock_hash = mc_node.generate(1)[0]
+        scblock_id3 = generate_next_blocks(sc_node1, "first node", 1)[0]
+        check_mcreference_presence(we1_1_mcblock_hash, scblock_id3, sc_node1)
+
+        # Wait until Certificate will appear in MC node mempool
+        time.sleep(10)
+        while mc_node.getmempoolinfo()["size"] < 1 and sc_node1.submitter_isCertGenerationActive()["result"]["state"]:
+            logging.info("Wait for certificate in mc mempool...")
+            time.sleep(2)
+        assert_equal(1, mc_node.getmempoolinfo()["size"], "Certificate was not added to Mc node mempool.")
+
+        # generate one MC block for certificate to appear in MC
+        # generate one SC block to sync with MC
+        we3_end_mcblock_hash = mc_node.generate(1)[0]
+        scblock_id3 = generate_next_block(sc_node1, "first node")
+        check_mcreferencedata_presence(we3_end_mcblock_hash, scblock_id3, sc_node1)
+
+        node_info_1 = sc_node1.node_info()['result']
+        node_info_2 = sc_node2.node_info()['result']
+
+        # sc node 1
+        assert_true(node_info_1['nodeName'], 'node0')
```

Comment:
Can we use when possible the data coming from the test class instead of hardcoding them please? (Same for UTXO test)

Reply:
Done, I was able to extract some info from `self`